src/command_modules/azure-cli-cosmosdb/azure/cli/command_modules/cosmosdb/actions.py

```diff
@@ -0,0 +1,54 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import argparse
+
+from knack.util import CLIError
+from knack.log import get_logger
+
+from azure.mgmt.cosmosdb.models import (
+    Location
+)
+
+logger = get_logger(__name__)
+
+
+# pylint: disable=line-too-long
+class CreateLocation(argparse._AppendAction):
+    def __call__(self, parser, namespace, values, option_string=None):
+        if namespace.locations is None:
+            namespace.locations = []
+
+        if any("regionname" in s.lower() for s in values):
+            keys_found = set()
+            _name = ""
+            _failover = 0
+            _is_zr = False
+            for item in values:
+                kvp = item.split('=', 1)
+                _key = kvp[0].lower()
+                if _key in keys_found:
+                    raise CLIError('usage error: --locations [KEY=VALUE ...]')
+                keys_found.add(_key)
+                if _key == "regionname":
+                    _name = kvp[1]
+                elif _key == "failoverpriority":
+                    _failover = int(kvp[1])
+                elif _key == "iszoneredundant":
+                    _is_zr = kvp[1].lower() == "true"
+                else:
+                    raise CLIError('usage error: --locations [KEY=VALUE ...]')
+            namespace.locations.append(
+                Location(location_name=_name, 
+                         failover_priority=_failover, 
+                         is_zone_redundant=_is_zr))
+        else:
+            logger.warning('DEPRECATION WARNING: The regionName=failoverPriority method of specifying locations is deprecated. Use --locations [KEY=VALUE ...] to specify the regionName, failoverPriority, and isZoneRedundant properties of the location. Multiple locations can be specified by including more than one --locations argument.')
```

Comment:
Consider here that if someone writes `--location eastus=1 --location westus=2` then they will see this warning twice. You might want to track a private variable on the namespace to track whether the warning has been displayed once and not display it again.  If you are okay with bugging the user to move to the new format, then this is fine. :)